password_manager_server_updated.py

```diff
@@ -0,0 +1,387 @@
+        username_to_app = data[j+2:]
+        print("site name: " + site_name)
+        print("username to app: " + username_to_app)
+        url = site_name
+        extracted_info = tldextract.extract(url)
+        site = site_name + ".txt"
+        print("site " + site)
+        file_path = "C:\\Users\\Ella Menachemi\\OneDrive\\Desktop\\password_manager\\" + username_to_app + "\\" + site
+        file = pathlib.Path(file_path)
+
+        if not file.exists():
+            reply = "The website you entered is not saved in the system"
+
+        else:
+            path = ("C:\\Users\\Ella Menachemi\\OneDrive\\Desktop\\password_manager\\" + username_to_app + "\\" + site)
+            file = open(path, "r")
+            content = file.readlines()
+            password_line = content[1]
+            password = password_line[10:]
+            reply = "site in system" + password
+
+        return reply
+
+    def check_password_validity(self, password):
+        is_valid = True
+        small = False
+        capital = False
+        special_char = False
+        for letter in password:
+            if letter.islower():
+                small = True
+            elif letter.isupper():
+                capital = True
+            if not letter.isdigit() and not letter.isupper() and not letter.islower():
+                special_char = True
+        if small and capital and special_char == False:
+            reply = "The password must contain at least one capital letter,\n one small letter and one special letter"
+            is_valid = False
+        if len(password) < 8:
+            reply = "password not long enough"
+            is_valid = False
+        else:
+            i = 0
+            count = 0
+            while i != len(password) -2:
+                if password[i].isdigit() and password[i+1].isdigit():
+                    if int(password[i]) == int(password[i+1]) + 1:
+                        count = count + 1
+                    elif int(password[i]) == int(password[i+1]) - 1:
+                        count = count+1
+                    else:
+                        count = 0
+                if count == 4:
+                    reply = "The password contains illegal series of chars"
+                    is_valid = False
+                    break
+                i = i + 1
+
+                if "asdf" in password:
+                    reply = "The password contains illegal series of chars"
+                    is_valid = False
+                if "qwerty" in password:
+                    reply = "The password contains illegal series of chars"
+                    is_valid = False
+
+
+        if is_valid:
+            reply = "password good"
+
+        return reply, is_valid
+
+    def check_username_sign_up(self, data):
+        data = data[18:]
+        self.username_to_app = data
+        print("username:"+self.username_to_app)
+        file_path = "C:\\Users\\Ella Menachemi\\OneDrive\\Desktop\\password_manager\\" + data
```

Comment:
Make the root directory of the files for the server configurable, and save as a member of the class

Reply:
Also use pathlib functionality (/ as a divider for folders) to make the code more clean